test/Google.Api.Gax.Grpc.Tests/ApiCallTest.cs

```diff
@@ -116,5 +117,51 @@ private void AssertMetadata(Metadata expected, Metadata actual)
         /// </summary>
         private static Metadata CreateMetadata(string key, string value) =>
             new Metadata { new Metadata.Entry("kind", "response") };
+
+        [Theory, CombinatorialData]
+        public async Task WithExceptionCustomizer_NoException(bool sync)
+        {
+            var response = new SimpleResponse();
+            var call = FakeApiCall.Create<SimpleRequest, SimpleResponse>((req, options) => response);
+            bool called = false;
+            call = call.WithExceptionCustomizer(original => { called = true; return null; });
+            var request = new SimpleRequest();
+            var actualResponse = sync ? call.Sync(request, null) : await call.Async(request, null);
+            Assert.Same(response, actualResponse);
+            Assert.False(called);
+        }
+
+        [Theory, CombinatorialData]
+        public async Task WithCustomExceptions_UncustomizedException(bool sync)
+        {
+            var response = new SimpleResponse();
+            var original = new RpcException(Status.DefaultCancelled);
+            var call = FakeApiCall.Create<SimpleRequest, SimpleResponse>((req, options) => throw original);
+            RpcException callbackException = null;
+            call = call.WithExceptionCustomizer(x => { callbackException = x; return null; });
+            var request = new SimpleRequest();
+            // It's slightly awkward to call specify the lambda in ThrowsAsync.
```

Comment:
Nit: Slight odd grammar in this comment...

Reply:
Have modified it somewhat. Will merge with the new version, and we can fiddle more later.